core/src/main/java/com/ritense/valtimo/service/UserSettingsService.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ *  Licensed under EUPL, Version 1.2 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.ritense.valtimo.service;
+
+import com.ritense.valtimo.contract.authentication.ManageableUser;
+import com.ritense.valtimo.domain.user.UserSettings;
+import com.ritense.valtimo.repository.UserSettingsRepository;
+
+import java.util.Optional;
+
+public class UserSettingsService {
+
+    private final UserSettingsRepository userSettingsRepository;
+
+    public UserSettingsService(UserSettingsRepository userSettingsRepository){
+        this.userSettingsRepository = userSettingsRepository;
+    }
+
+    public Optional<UserSettings> getCurrentUserSettings(ManageableUser currentUser) {
+        return userSettingsRepository.findById(currentUser.getId());
+    }
+
+    public void upsertUserSettings(ManageableUser currentUser, String settings) {
```

Comment:
I've not seen 'upsert' being used anywhere in Valtimo. Please change to `saveUserSettings`